.gitignore

```diff
@@ -62,7 +62,6 @@ json_formatter
 *.vcxproj.filters
 *.lastbuildstate
 .vs/
-.vscode/
```

Comment:
What if there are some other unrelated files in that folder?

Reply:
There are a few options:

1. I can whitelist only the currently required files (I believe that's possible with .gitignore). 
2. I can make it into a multifolder workspace which uses a separate file in the root of the directory, as mentioned in the OP.

Reply:
I went with option one. The entire folder is ignored and we will explicitly allow that which should be checked in.